frontend/src/components/Map.tsx

```diff
@@ -0,0 +1,467 @@
+                },
+                logEvent: 'mlc',
+                logMapZoom: map.getZoom(),
+                context: startup.context,
+            }).then((res) => res.json() as Promise<StackResponse>);
+        },
+        enabled:
+            !!expression &&
+            !!startup?.session &&
+            map?.getCenter() &&
+            map?.getZoom() !== undefined,
+    });
+
+    useEffect(() => {
+        if (!stackQuery.data) return;
+        const { proto, geoJSON } = stackQuery.data;
+        if (proto && coordinates && expression) {
+            setAppAtom((draft) => {
+                draft.stacks = pickBy(
+                    stacks,
+                    (stack) => !stack.transient || stack.docked
+                );
+                draft.stacks[proto.expression] = {
+                    coordinates,
+                    docked: false,
+                    expression: expression,
+                    proto,
+                    id: proto.expression,
+                    transient: true,
+                };
+                draft.geojson[proto.expression] = geoJSON;
+            });
+        }
+    }, [stackQuery.data]);
+
+    const handleClick = useCallback(
+        (evt: MapLayerMouseEvent) => {
+            setCoordinates(evt.point);
+            setExpression(
+                `${evt.lngLat.lat.toFixed(6)}, ${evt.lngLat.lng.toFixed(6)}`
+            );
+            /* const feature = evt.features?.[0];
+        if (!feature) return;
+        const { ns, id } = feature.properties;
+        if (!ns || !id) return; */
+
+            /* setAppAtom((draft) => {
+                draft.stacks[stackId] = {
+                    coordinates: evt.point,
+                    expression: `${evt.lngLat.lat.toFixed(
+                        6
+                    )}, ${evt.lngLat.lng.toFixed(6)}`,
+                };
+            }); */
+
+            return;
+        },
+        [setCoordinates, setExpression]
+    );
+
+    const draggableStacks = useMemo(() => {
+        return pickBy(stacks, (stack) => !stack.docked);
+    }, [stacks]);
+
+    const dockedStacks = useMemo(() => {
+        return pickBy(stacks, (stack) => stack.docked);
+    }, [stacks]);
+
+    const mvt = new MVTLayer({
+        data: ['api/tiles/base/{z}/{x}/{y}.mvt'],
+        minZoom: 10,
+        maxZoom: 16,
+        /* getLineColor: (f: Feature) => {
+            return 'transparent';
+        }, */
+        getFillColor: (f: Feature) => {
+            if (f.properties.layerName === 'query') {
+                //console.log(f);
+            }
+            // console.log(f);
+            return 'transparent';
+        },
+
+        //onDataLoad: (data: $FixMe) => console.log(data),
+    });
+
+    const layers = [mvt];
+    console.log(layers);
+
+    /* 
+    not ideal that we're transforming to array and filtering such a large dataset here on every render. @TODO: improve performance 
+    */
+    const points = useMemo(() => {
```

Comment:
Good question, for two reasons: 1. We're rendering these as Markers on top of the map, so these are SVG elements instead. I think this can be useful for custom interactions and allows us to customize the icons and such. It seems doable performance-wise as there aren't an awful lot of them at the same time in the viewport. This takes us to 2 - for it to be feasible, we're filtering out the points outside the viewport (and I also noticed we have a few duplicates for the same coordinates, so we're getting rid of those also).